src/iceberg/file_format.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+/// \file iceberg/file_format.h
+/// File format used by Iceberg.
+
+#include <string_view>
+
+#include "iceberg/iceberg_export.h"
+
+namespace iceberg {
+
+/// \brief File format type
+enum class ICEBERG_EXPORT FileFormatType {
+  kParquet,
+  kAvro,
+  kOrc,
+  kPuffin,
+  kMetadata,
```

Comment:
Should this be kJson? Metadata is a file format to me ;(

Reply:
Hug, I just checked the java impl, kMetadata should be fine, sorry for the noise.

Reply:
The `kMetadata` confuses me a bit. I asssume that would mean Iceberg table metadata (which is encoded in Avro).

Wouldn't it make more sense to align this with the `file_format` field:

![image](https://github.com/user-attachments/assets/5d8fee05-9d14-49ae-a500-7c4e5d225bca)

Reply:
Make sense. I think we don't have to copy what the java impl does. Removed `kMetadata`.